frontend/src/app/(vendor)/vendors/login/page.tsx

```diff
@@ -0,0 +1,125 @@
+import { fetchVendorBySlug } from "@/features/profile/common/api/fetchVendor";
+import { ReCaptchaRef } from '@/components/security/ReCaptcha';
+import { useNotification } from "@/contexts/NotificationContext";
+
+export default function VendorLoginPage() {
+  const { addNotification } = useNotification();
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const recaptchaRef = useRef<ReCaptchaRef>(null);
+
+  // Extract magic link parameters
+  const slug = searchParams.get('slug') || "";
+  const email = searchParams.get('email') || "";
+  const token = searchParams.get('token') || "";
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (process.env.NEXT_PUBLIC_FEATURE_VENDOR_LOGIN_ENABLED !== 'true') {
+      router.push(`/`);
+      return;
+    }
+
+    // Check if user is already logged in
+    const checkSession = async () => {
+      const supabase = createClient();
+      const { data: { session } } = await supabase.auth.getSession();
+
+      if (session) {
+        console.log("User is already logged in");
+        // User is already logged in, redirect to vendor page
+        router.push(`/`);
+        return;
+      }
+
+      const areAllParamsValid = !!email && !!token && email.trim() !== "" && token.trim() !== "" && slug.trim() !== "";
+
+      if (!areAllParamsValid) {
+        // Email, token, or slug is missing, show the login form
+        setIsLoading(false);
+        return;
+      }
+
+      // Execute reCAPTCHA and get token
+      try {
+        await recaptchaRef.current?.executeAsync();
+        console.log("reCAPTCHA executed successfully");
+      } catch (error) {
+        console.error("Error executing reCAPTCHA: ", error);
+        setIsLoading(false);
+        return;
+      }
+
+      // Check if email and token from the query parameters are valid and match database records. 
+      // If they do, sign in the user anonymously and link their email to the account.
+      const vendor = await fetchVendorBySlug(slug);
+      const doEmailAndTokenMatch = email === vendor?.email?.toLowerCase() && token === vendor?.access_token?.toLowerCase();
+
+      if (doEmailAndTokenMatch) {
+        // Sign in the user anonymously and link email
+        const { error } = await supabase.auth.signInAnonymously()
+        if (error) {
+          console.error("Error signing in anonymously: " + error.message);
+          setIsLoading(false);
+          return;
+        }
+
+        const { data: updateEmailData, error: updateEmailError } = await supabase.auth.updateUser({
+          email,
+        })
+
+        if (updateEmailError) {
+          console.error("Error updating user email: " + updateEmailError.message);
+          setIsLoading(false);
+          return;
+        }
+        console.log("Successfully logged in and updated user email for user ID: " + updateEmailData.user?.id);
+        router.push(`/vendors/${slug}`);
```

Comment:
fyi this will go to a dynamic `/manage` page, not a static page. I can update the route with my changes.

Since it's dynamic, is there an easy way to look up the vendor from the user? Through the access token?

Reply:
we can look it up through the access token or keep the slug as a url param!